### ISSUE TITLE
Incorrect branch when running on a tag push commit?

### ISSUE THREAD

**Issue by HarelM** [appveyor/ci#2499]
See the following image:
![image](https://user-images.githubusercontent.com/3269297/42418597-7080717a-82ac-11e8-82f0-95c1ef868c5b.png)

https://ci.appveyor.com/project/IsraelHikingHost/site/build/7.0.81

I don't have a branch called v7.0 I only pushed a tag to master branch in order to create a release with artifacts in github.

There's also a warning at the end of the log file which tries to check if this is master branch but fails and does not copy artifacts:
```
"GitHub" deployment for branch "master" has been skipped because current branch is "v7.0"
Build success
```

I will remove the condition to deploy only when building master as a workaround, I hope it will be fixed soon so I can add this condition back.

Project github repo:
https://github.com/IsraelHikingMap/Site

What I would like to achieve, and I'm not sure if this is possible to do, is the following:
1. Create a tag
2. Push it to remote
3. Cause Appveyor to generate artifacts and copy them to GitHub
4. Rename the tag and release version to match the version create automatically by Appveyor.

**Comment by IlyaFinkelshteyn** [comment]
I do not think it is a bug. This is expected and well known behavior which is documented [here](https://www.appveyor.com/docs/deployment/#deploy-on-tag-github-and-gitlab-only) (check part related to **annotated** tags) and was discussed, for example, [here](https://github.com/appveyor/ci/issues/898).

Side note:  when you tune GitHub deployment on tag, add this setting to GitHub deployment:`tag: $(APPVEYOR_REPO_TAG_NAME)`. It will prevent you from infinite loop when AppVeyor will create new tag on each deployment and therefore start new tagged build.

I will keep this issue open, but fix will be adding more self-descriptive warning about skipped deployment (and build in case of build filtering) to avoid confusion.

**Comment by HarelM** [comment]
Take a look at the screen shot - The branch name is v7.0 - I don't have a branch with that name - this is the main issue here, from my perspective.
It might be a duplicate of #2499 though.

**Comment by IlyaFinkelshteyn** [comment]
@HarelM: we agree. We change both UI part and add more self-explaining messages.  Please confirm that you are unblocked though. Also what is a duplicate? Seems that you placed `this` issue link here.

**Comment by HarelM** [comment]
I'm not blocked. I have changed the yml to work on all branches and not just master so it's working now.
See here: https://github.com/IsraelHikingMap/Site/blob/master/appveyor.yml
>It might be a duplicate of #2499 though.

I meant to say that this issue might be a duplicate of #2499, I'm not sure though.
